### ISSUE TITLE
"The CLI tools work with .NET Core projects and .NET Framework projects" error

### ISSUE THREAD

**Issue by jeubank12** [aspnet/EntityFramework.Docs#1228]
This needs specific documentation on how to make this work.

A command such as `dotnet ef migrations script --framework 472 -p net472project\net472project.csproj -s net472project\net472project.csproj` results in the following error: 
"error MSB4019: The imported project "C:\Program Files\dotnet\sdk\2.1.503\Microsoft\VisualStudio\v15.0\WebApplications\Microsoft.WebApplication.targets" was not found. Confirm that the path in the &lt;Import&gt; declaration is correct, and that the file exists on disk.
Unable to retrieve project metadata. Ensure it's an MSBuild-based .NET Core project. If you're using custom BaseIntermediateOutputPath or MSBuildProjectExtensionsPath values, Use the --msbuildprojectextensionspath option."

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 263f21d6-6edc-7582-e0e2-fa1c9670bcc0
* Version Independent ID: b3dd8c84-18b1-f98d-64f1-7897d1482692
* Content: [EF Core tools reference (.NET CLI) - EF Core](https://docs.microsoft.com/en-us/ef/core/miscellaneous/cli/dotnet)
* Content Source: [entity-framework/core/miscellaneous/cli/dotnet.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/miscellaneous/cli/dotnet.md)
* Product: **entity-framework**
* GitHub Login: @bricelam
* Microsoft Alias: **bricelam**

**Comment by Moorepheus** [comment]
Having a similar issue.  Have a .NetFramework project (as start up), but created a dummy .Net Core project to run CLI, but I get this error"

C:\Users\XXXXX\Documents\Visual Studio 2017\Projects\efcoretest2\obj\EFCoreTest2.csproj.EntityFrameworkCore.targets(4,5): error MSB4006: There is a circular dependency in the target dependency graph involving target "GetEFProjectMetadata". [C:\Users\XXXXX\Documents\Visual Studio 2017\Projects\efcoretest2\EFCoreTest2.csproj]
Unable to retrieve project metadata. Ensure it's an MSBuild-based .NET Core project. If you're using custom BaseIntermediateOutputPath or MSBuildProjectExtensionsPath values, Use the --msbuildprojectextensionspath option."

Not sure what I am missing.

**Comment by Davy-F** [comment]
I'm also getting this issue with .NET Core build during the .NET Core Build pipeline.

`
  Nothing to do. None of the projects specified contain packages to restore.
D:\a\1\s\...\packages\Microsoft.Web.WebJobs.Publish.2.0.0\build\webjobs.console.targets(24,3): error MSB4019: The imported project "C:\Program Files\dotnet\sdk\2.2.105\Microsoft\VisualStudio\v15.0\WebApplications\Microsoft.WebApplication.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [D:\a\1\s\....csproj]

Build FAILED.
`

**Comment by L0G1C** [comment]
I'm also seeing this issue just trying to create a script by running migrations with EFcore in a project I created separate from my Asp.Net Core Web project. The web project properties lists it as a .netcore 2.2 project, but the new project I made is .net 4.7.1. When I try this dotnet ef comman din Azure build it fails with "Unable to retrieve project metadata. Ensure it's an MSBuild-based .NET Core project"
